sources/packages/api/src/route-controllers/institution/institution.controller.service.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  Injectable,
+  NotFoundException,
+  BadRequestException,
+} from "@nestjs/common";
+import { InstitutionService, FormService } from "../../services";
+import {
+  INSTITUTION_TYPE_BC_PRIVATE,
+  getExtendedDateFormat,
+} from "../../utilities";
+import {
+  InstitutionReadOnlyDTO,
+  InstitutionContactDTO,
+  InstitutionProfileDTO,
+} from "./models/institution.dto";
+import { FormNames } from "../../services/form/constants";
+
+/**
+ * Service/Provider for Institutions controller to wrap the common methods.
+ */
+@Injectable()
+export class InstitutionControllerService {
+  constructor(
+    private readonly institutionService: InstitutionService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Get institution detail.
+   * @param institutionId
+   * @param clientType
+   * @returns InstitutionReadOnlyDTO
+   */
+  async getInstitutionDetail(
+    institutionId: number,
+  ): Promise<InstitutionReadOnlyDTO> {
+    const institutionDetail =
+      await this.institutionService.getInstitutionDetailById(institutionId);
+
+    if (!institutionDetail) {
+      throw new NotFoundException("Institution not valid");
```

Comment:
The same comment as given in the previous file, NotFoundException does not align with the comment given

Reply:
Here the institutionId is a path param, so an invalid institutionId should lead to 404. That's why not found was used.